map not loading when online

pradeepmdk opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/26589
PROPOSAL: https://github.com/Expensify/App/issues/26589


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Make sure to log out.
2. Refresh the page (for browser) or reopen the app (for native/desktop app).
3. Log in with any account.
4. Make sure device offline.
5. Click plus button and then select Distance Request.
6. Switch the device back online.
7. Verify the map displays properly.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Make sure to log out.
2. Refresh the page (for browser) or reopen the app (for native/desktop app).
3. Log in with any account.
4. Make sure device offline.
5. Click plus button and then select Distance Request.
6. Switch the device back online.
7. Verify the map displays properly.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/16595846/e495fbeb-9a57-470c-8628-2bffd6978f9d


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16595846/1404930b-7448-4d42-9ef2-c3d9efde942d



</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16595846/653c2a1a-7481-4b95-bae5-8ba0b5967acf



</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16595846/3f669550-d87c-45aa-b545-6f75946f71f3


</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16595846/52070419-7762-4096-9d1c-7ad983326662


</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16595846/96632b1e-0cf8-4a17-8e59-b90420c08547


</details>

pradeepmdk committed:
map loding when online

pradeepmdk committed:
load chunk on desktop

pradeepmdk commented on src/setup/platformSetup/index.website.js:
```diff
@@ -3,6 +3,10 @@ import {AppRegistry} from 'react-native';
 import 'shim-keyboard-event-key';
 
+// load all chunk file map when online
+import 'react-map-gl';
```
@ntdiary do you have any suggestion here where we need load. when I am checking https://github.com/Expensify/App/blob/d89ca7043b9bc923c0599836a3571bb15d40ed7c/src/components/MapView/MapView.web.tsx#L8C36-L8C48 loading this but when we open distance page at the time only components mounted so it's loading all necessary chunk file when offline this is ignored.

Note: This will be needed only for the web and desktop. so I added on the platform setup this will call initially and load all chunk files. like shim-keyboard-event-key

pradeepmdk replied:
@ntdiary could you check this

pradeepmdk replied:
@ntdiary  MapView.web.tsx already we have the import statement. the actual problem is MapView.web.tsx will trigger only when you open a distance request page like a token api trigger. before open up the distance request page your offline means map file(chunk) is not loading.

pradeepmdk replied:
@ntdiary this is working  `mapLib={import('mapbox-gl')}` let me update PR

hayata-suenaga commented:
hi @ntdiary @pradeepmdk 

I'm sorry I'm juggling several issues in addition to a different project

could you tell me what chunk files are and why we need to touch file import statements in this PR?

pradeepmdk commented:
> hi @ntdiary @pradeepmdk
> 
> I'm sorry I'm juggling several issues in addition to a different project
> 
> could you tell me what chunk files are and why we need to touch file import statements in this PR?

https://github.com/Expensify/App/assets/16595846/b33aa7dc-8344-4268-a2cf-35fc21f5a327

@hayata-suenaga when offline the first time opening the distance request and after that when try open multiple times map doesn't load.

pradeepmdk committed:
map view updated

hayata-suenaga commented:
@ntdiary 

thank you so much for the explanation. could you explain more about these two options?

> we either need to bundle the mapbox-gl package into the vendors-xxx.js file, or we need to dynamically load mapbox-gl in MapView.web.tsx using import().

hayata-suenaga commented:
ah I see that made thing more clear

@ntdiary which do you think is the best option? I think dynamic import seems better to me?

pradeepmdk commented:
@ntdiary @hayata-suenaga dynamic import is working fine in our project. I already updated the PR

hayata-suenaga commented:
> Besides the dynamic import has an error log for the first offline load

Is there a way to suppress the error log? why do we get the error log? (sorry for asking a lot of questions)

> Also, we haven't used dynamic import in our project yet

I thought we're using dynamic imports in some of the screens (we only load/fetch components code when navigated to them?)

pradeepmdk commented:
@hayata-suenaga bump

hayata-suenaga commented:
ah I thought that separate files are fetched with module imports. Let's go with module import approach

@pradeepmdk sorry for the late response 🙇

pradeepmdk committed:
update module import

pradeepmdk commented:
@hayata-suenaga np. update now  to module import @ntdiary

pradeepmdk committed:
move logic to maptoken

pradeepmdk commented:
> @pradeepmdk, There are two pending tasks need to be addressed, feel free to let me know if there are any problems. : )

sorry @ntdiary i am not able to see this in review comments so that missed this. i am updated now.. 
let me know if you want ignore this line here 
`    const doesRouteExist = !lodashIsNull(lodashGet(transaction, 'routes.route0.geometry.coordinates'));
`
i will remove it.

pradeepmdk commented:
@ntdiary updated.

pradeepmdk commented:
> @pradeepmdk, great! Could you please explain this? And we are very close to the end. 😄 <img alt="image" width="565" src="https://user-images.githubusercontent.com/8579651/268804513-5d28adad-47c3-4304-9d55-3b2258f28b7d.png">

yes i agree we don't need when i am testing i removed the token from indexdb manually  deleted so callback its not triggering so that i added this one.


@ntdiary one small doubt how you are adding this review comment because i am not able see this here.🤔

pradeepmdk committed:
remove unwanted code

pradeepmdk committed:
update function

ntdiary reviewed (approved):
@hayata-suenaga, all yours! : )

pradeepmdk commented:
@ntdiary updated

pradeepmdk commented:
@hayata-suenaga,  bump;

hayata-suenaga commented on src/libs/actions/MapboxToken.js:
```diff
@@ -46,11 +46,16 @@ const hasTokenExpired = () => moment().isAfter(currentToken.expiration);
 };
 
+const fetchToken = () => {
+    console.debug('[MapboxToken] Token does not exist so fetching one');
```
should we keep this one?

pradeepmdk replied:
@hayata-suenaga you mean console log or this `fetchToken` method?

hayata-suenaga replied:
yes I don't think we have debug logs in our codebase?

pradeepmdk replied:
we have a debug log for onyx updates, pusher events, network queue

pradeepmdk replied:
let me know if you want to remove here let me update.

hayata-suenaga replied:
@ntdiary can you confirm ⬆️ ?

hayata-suenaga replied:
ah I don't think no, but we can remove this one right?

pradeepmdk replied:
updated @hayata-suenaga

pradeepmdk committed:
remvoe console

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6342877713) to staging by https://github.com/hayata-suenaga in version: 1.3.75-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6383226401) to production by https://github.com/mountiny in version: 1.3.75-12 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6383232884) to staging by https://github.com/hayata-suenaga in version: 1.3.76-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6395179660) to production by https://github.com/mountiny in version: 1.3.76-6 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅
